homeassistant/components/homeconnect/config_flow.py

```diff
@@ -0,0 +1,28 @@
+"""Config flow for Home Connect."""
+import logging
+
+from homeassistant import config_entries
+from homeassistant.helpers import config_entry_oauth2_flow
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class OAuth2FlowHandler(config_entry_oauth2_flow.AbstractOAuth2FlowHandler):
+    """Config flow to handle Home Connect OAuth2 authentication."""
+
+    DOMAIN = DOMAIN
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_PUSH
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        if self.hass.config_entries.async_entries(DOMAIN):
```

Comment:
Why limit it to 1 installation? It looks like the code can handle multiple config entries ?

Reply:
You're right, I never actually thought about this, I will look into it!

Reply:
Just to be sure I understood correctly: I can just get rid of `async_step_user` altogether and that's it, right? Did that in https://github.com/home-assistant/home-assistant/pull/29214/commits/1b88efb77e83bd60a1c2b65992f7ed4b14af9782.